src/django_twc_toolbox/crud/views.py

```diff
@@ -52,3 +53,68 @@ def get_detail_fields(self):
 
     def get_list_fields(self):
         return self.list_fields
+
+    @override
+    def list(
+        self, request: HttpRequest, *args: object, **kwargs: object
+    ) -> TemplateResponse:
+        """GET handler for the list view."""
+
+        queryset = self.get_queryset()
+        filterset = self.get_filterset(queryset)
+        if filterset is not None:
+            queryset = filterset.qs
+
+        if not self.allow_empty and not queryset.exists():
+            raise Http404
+
+        if self.table_class is not None:
+            paginate_by = self.get_paginate_by(queryset)  # type: ignore[call-arg,reportCallIssue,reportUnknownVariableType]
+        else:
+            paginate_by = self.get_paginate_by()
+
+        if paginate_by is None:
+            # Unpaginated response
+            self.object_list = queryset
+            context = self.get_context_data(
+                page_obj=None,
+                is_paginated=False,
+                paginator=None,
+                filterset=filterset,
+            )
+        else:
+            # Paginated response
+            page = self.paginate_queryset(queryset, paginate_by)
+            self.object_list = page.object_list
+            context = self.get_context_data(
+                page_obj=page,
+                is_paginated=page.has_other_pages(),
+                paginator=page.paginator,
+                filterset=filterset,
+            )
+
+        return self.render_to_response(context)
+
+    @override
+    def get_context_data(self, **kwargs: object) -> dict[str, object]:
+        context = super().get_context_data(**kwargs)
+        context["list_view_url"] = Role.LIST.maybe_reverse(self)
+        if self.object is not None:
+            context["delete_view_url"] = Role.DELETE.maybe_reverse(self, self.object)
+            context["detail_view_url"] = Role.DETAIL.maybe_reverse(self, self.object)
+            context["update_view_url"] = Role.UPDATE.maybe_reverse(self, self.object)
+        return context
+
+    @classmethod
+    @override
+    def as_view(  # type: ignore[override]
+        cls, role: Role, **initkwargs: object
+    ) -> Callable[..., HttpResponse]:
+        if role != Role.LIST or cls.table_class is None:
+            return super().as_view(role=role, **initkwargs)
+
+        class ListViewWithTable(SingleTableMixin, cls): ...  # type: ignore[misc,valid-type]
+
+        return ListViewWithTable.as_view(
+            role=role, table_class=cls.table_class, **initkwargs
+        )
```

Comment:
This could use some notes / comments. I was scratching my head until I got to here what was going on.